clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -15629,6 +15629,34 @@ ExprResult SemaOpenMP::VerifyPositiveIntegerConstantInClause(
   return ICE;
 }
 
+void SemaOpenMP::ActOnOpenMPDeviceNum(Expr *DeviceNumExpr) {
+  llvm::APSInt Result;
+  Expr::EvalResult EvalResult;
+  // Strip implicit casts from the expression
+  DeviceNumExpr = DeviceNumExpr->IgnoreImpCasts();
```

Comment:
Why need to drop the casts here?

Reply:
I was unsure whether `EvaluateAsInt` could fetch the value from the expression containing implicit casts as the control kept ending up in the else case.

Reply:
Tru not to drop it

Reply:
Done! Thanks for the suggestion. Dropping casts wasn't necessary.